tests/models/test_licensing.py

```diff
@@ -502,195 +650,255 @@ def test_update_availability_does_nothing_if_given_no_data(self):
-        assert 40 == pool.patrons_in_hold_queue
+        assert pool.licenses_owned == 5
+        assert pool.licenses_available == 20
+        assert pool.licenses_reserved == 30
+        assert pool.patrons_in_hold_queue == 40
 
-
-    def test_open_access_links(self):
-        edition, pool = self._edition(with_open_access_download=True)
-        source = DataSource.lookup(self._db, DataSource.GUTENBERG)
+    def test_open_access_links(self, db_session, create_edition):
+        """
+        GIVEN: A LicensePool with an open access hyperlink Identifier
+        WHEN:  Adding two hyperlink Identifiers, one with open access and one without
+               and checking the LicensePool's open access links
+        THEN:  The two open access links are correctly identified
+        """
+        _, pool = create_edition(db_session, with_open_access_download=True)
+        source = DataSource.lookup(db_session, DataSource.GUTENBERG)
 
         [oa1] = list(pool.open_access_links)
 
         # We have one open-access download, let's
         # add another.
-        url = self._url
+        url = "https://example.com"
         media_type = MediaTypes.EPUB_MEDIA_TYPE
-        link2, new = pool.identifier.add_link(
+        link2, _ = pool.identifier.add_link(
             Hyperlink.OPEN_ACCESS_DOWNLOAD, url,
             source, media_type
         )
         oa2 = link2.resource
 
         # And let's add a link that's not an open-access download.
-        url = self._url
-        image, new = pool.identifier.add_link(
+        url = "https://example.com"
+        pool.identifier.add_link(
             Hyperlink.IMAGE, url, source, MediaTypes.JPEG_MEDIA_TYPE
         )
-        self._db.commit()
+        db_session.commit()
 
         # Only the two open-access download links show up.
         assert set([oa1, oa2]) == set(pool.open_access_links)
 
-    def test_better_open_access_pool_than(self):
-
-        gutenberg_1 = self._licensepool(
-            None, open_access=True, data_source_name=DataSource.GUTENBERG,
+    def test_better_open_access_pool_than(
+            self, db_session, create_edition, create_licensepool, init_datasource_and_genres):
+        """
+        GIVEN: An assortment of LicensePools
+        WHEN:  Determining if open-access pool is generall known for better-quality than another pool
+        THEN:  Correct determination
+        """
+        def get_licensepool(open_access, data_source_name, with_open_access_download=False, id=None):
+            """
+            Helper function to generate license pools.
+            """
+            edition = create_edition(db_session, identifier_id=id)
+            pool = create_licensepool(
+                db_session,
+                edition,
+                open_access=open_access,
+                data_source_name=data_source_name,
+                with_open_access_download=with_open_access_download
+            )
+            return pool
```

Comment:
I believe you mentioned this in our call. I'd like to find a way to make this a real fixture but this is OK for now.